src/helpers/issue.ts

```diff
@@ -160,6 +160,73 @@ export const getAllIssueComments = async (issue_number: number): Promise<Comment
   return result;
 };
 
+export const getAllIssueAssignEvents = async (issue_number: number): Promise<AssignEvent[]> => {
+  const context = getBotContext();
+  const payload = context.payload as Payload;
+
+  const result: AssignEvent[] = [];
+  let shouldFetch = true;
+  let page_number = 1;
+  try {
+    while (shouldFetch) {
+      const response = await context.octokit.rest.issues.listEvents({
+        owner: payload.repository.owner.login,
+        repo: payload.repository.name,
+        issue_number: issue_number,
+        per_page: 100,
+        page: page_number,
+      });
+
+      await checkRateLimitGit(response?.headers);
+
+      // Fixing infinite loop here, it keeps looping even when its an empty array
+      if (response?.data?.length > 0) {
+        response.data.filter((item) => item.event === "assigned").forEach((item) => result?.push(item as AssignEvent));
+        page_number++;
+      } else {
+        shouldFetch = false;
+      }
+    }
+  } catch (e: unknown) {
+    shouldFetch = false;
+  }
```

Comment:
I can't seem to find the original issue that I filed (I quickly checked `@ubiquibot/production/issues` `@ubiquibot/staging/issues` and `@ubiquity/ubiquibot/issues`

There was a bug with the `@ubiquibot[qa] bot running inside of GitHub Actions that would time out because it would run for 6 hours occasionally. Was this infinite loop the source of that issue?

Reply:
This was just copy pasted from the function above so probably it's not the source of that issue